server/plugins/renderer/renderer.module.spec.js

```diff
@@ -158,7 +167,48 @@ describe('Renderer Plugin', () => {
                 'stencil-options': '{"get_template_file":true,"get_data_only":true}',
                 'stencil-version': PACKAGE_INFO.config.stencil_version,
                 'accept-encoding': 'identity',
+                'stencil-custom-templates':
+                    '{"brand":{"a":"/abc/"},"category":{},"page":{},"product":{}}',
+            });
+        });
+
+        it('should send a request to the storefront server with added custom layouts to headers', async () => {
+            const limitedServerOptions = {
+                dotStencilFile: {
+                    storeUrl,
+                    normalStoreUrl,
+                    port: 4001,
+                    username: 'testUser',
+                    token: '6832b1c755bb9de13aa8990216a69a7623043fd7',
+                },
+                useCache: false,
+                themePath: themeConfigManager.themePath,
+            };
+
+            const newServer = await Server.create(limitedServerOptions);
+
+            // Don't log errors during the test
+            newServer.ext('onPostHandler', (request, h) => {
+                if (request.response.isBoom) {
+                    return h.response().code(500);
+                }
+                return h.continue;
+            });
+
+            axiosMock.onPost().reply(301, undefined, storefrontResponseHeaders);
+            await newServer.inject(browserRequest);
+
+            expect(axiosMock.history.post[0].headers).toMatchObject({
+                'Content-Type': 'application/x-www-form-urlencoded',
+                host: new URL(storeUrl).host,
+                'stencil-cli': PACKAGE_INFO.version,
+                'stencil-options': '{"get_template_file":true,"get_data_only":true}',
+                'stencil-version': PACKAGE_INFO.config.stencil_version,
+                'accept-encoding': 'identity',
+                'stencil-custom-templates': '{"brand":{},"category":{},"page":{},"product":{}}',
```

Comment:
here we receive `'{"brand":{"a":"/abc/"},"category":{},"page":{},"product":{}}'` instead of `'{"brand":{},"category":{},"page":{},"product":{}}'`